### ISSUE TITLE
Support for Razer Basilisk X Hyperspeed

### ISSUE THREAD

**Issue by tonydang4610** [1kc/razer-macos#497]
Would love to see support for this mouse, thanks!

**Comment by brocktubre** [comment]
Yeah I just bought this mouse from Best Buy and I am sad that I can't change the DPI for this device.

**Comment by chifle-silva** [comment]
+1, please add support for this mouse. Thank you!

**Comment by WrathofthePast** [comment]
I can do it for you if you're interested in testing it. Openrazer didn't find the transaction.id yet, but has everything else needed. We can try brute force guessing like I did for #486. It's likely to be the same as another Basilisk mouse.

**Comment by tonydang4610** [comment]
> 

I'm definitely down to test for it.

**Comment by WrathofthePast** [comment]
Does it work with the dongle, but not bluetooth? It sounds like the dongle is the same as Basilisk Ultimate which is supported. Obviously, bluetooth support will need to be added, but that's a separate implementation.

**Comment by tonydang4610** [comment]
> Does it work with the dongle, but not bluetooth? It sounds like the dongle is the same as Basilisk Ultimate which is supported. Obviously, bluetooth support will need to be added, but that's a separate implementation.

I have not tried it with the dongle since I'm using a MacBook. But I know for sure there is no support for Bluetooth.

**Comment by WrathofthePast** [comment]
If you are able to try while I work on the bluetooth implementation, that'd be helpful. I'll be able to add bluetooth support without it, but I won't know if the dongle works or not.

**Comment by tonydang4610** [comment]
> If you are able to try while I work on the bluetooth implementation, that'd be helpful. I'll be able to add bluetooth support without it, but I won't know if the dongle works or not.

I really wish I could but I do not have a USB A adaptor on me. Sorry, wish I could've been more helpful.

**Comment by chifle-silva** [comment]
The mouse works with the dongle yes

**Comment by WrathofthePast** [comment]
> The mouse works with the dongle yes

@chifle-silva, thank you that's very helpful.

**Comment by WrathofthePast** [comment]
So while I'm working on this, I discover that it doesn't have RGB lightning. Can you connect via bluetooth without razer-macos support? I'm a little confused what's the to gain from adding support beyond DPI and Polling Rate control.

**Comment by tonydang4610** [comment]
> So while I'm working on this, I discover that it doesn't have RGB lightning. Can you connect via bluetooth without razer-macos support? I'm a little confused what's the to gain from adding support beyond DPI and Polling Rate control.

Having support for DPI and Polling Rate is enough for the mouse. I'd love to change the DPI on the mouse without doing so in macOS settings.

**Comment by WrathofthePast** [comment]
> Having support for DPI and Polling Rate is enough for the mouse. I'd love to change the DPI on the mouse without doing so in macOS settings.

So you're saying you're able to connect via bluetooth as is? I'll add support, regardless, but I need to make sure I understand what the starting point is.

**Comment by tonydang4610** [comment]
> > Having support for DPI and Polling Rate is enough for the mouse. I'd love to change the DPI on the mouse without doing so in macOS settings.
> 
> So you're saying you're able to connect via bluetooth as is? I'll add support, regardless, but I need to make sure I understand what the starting point is.

Yes, I can connect via Bluetooth as is.

**Comment by WrathofthePast** [comment]
One more question: Razer doesn't list the max polling rate of the mouse. Is it 1000Hz or 8KHz? I'm assuming the former as I can't imagine why they wouldn't advertise a 8KHz polling rate.

**Comment by tonydang4610** [comment]
1,000 Hz

**Comment by chifle-silva** [comment]
> > > Having support for DPI and Polling Rate is enough for the mouse. I'd love to change the DPI on the mouse without doing so in macOS settings.
> > 
> > 
> > So you're saying you're able to connect via bluetooth as is? I'll add support, regardless, but I need to make sure I understand what the starting point is.
> 
> Yes, I can connect via Bluetooth as is.

Same, I can connect via Bluetooth as is

**Comment by WrathofthePast** [comment]
I've run into a discrepancy with the information openrazer provided. I need a screenshot of its information under System Information/USB.

**Comment by chifle-silva** [comment]
<img width="886" alt="Screen Shot 2021-12-19 at 22 52 30" src="https://user-images.githubusercontent.com/8663001/146692131-1c295fe0-cf5a-4271-b59d-736baf3b3df6.png">


here you have. Let me know if that is what you need

**Comment by WrathofthePast** [comment]
> here you have. Let me know if that is what you need

@chifle-silva, yes, thank you. openrazer had the device-id as 0x0082 and that's why they never got it working.

**Comment by tonydang4610** [comment]
> > here you have. Let me know if that is what you need
> 
> @chifle-silva, yes, thank you. openrazer had the device-id as 0x0082 and that's why they never got it working.

So will there be support for this mouse?

**Comment by WrathofthePast** [comment]
Sorry, I've been busy. I finished the frontend & I'm 2/3 done the backend. I'll finish it in the very near future. Then, I'll create a test release for you to try.

**Comment by tonydang4610** [comment]
Sounds great. Thanks.

**Comment by CodeMan98038** [comment]
someone tell me if this works

**Comment by WrathofthePast** [comment]
> Sorry, I've been busy. I finished the frontend & I'm 2/3 done the backend. I'll finish it in the very near future. Then, I'll create a test release for you to try.

Sorry for the further delay. Code is complete & I'll provide a test build tonight or tomorrow (reinstalling MacOS right now for unrelated reasons, so I can't do it this second).

**Comment by CodeMan98038** [comment]
Sweet thanks a lot!

**Comment by WrathofthePast** [comment]
> > Sorry, I've been busy. I finished the frontend & I'm 2/3 done the backend. I'll finish it in the very near future. Then, I'll create a test release for you to try.
> 
> Sorry for the further delay. Code is complete & I'll provide a test build tonight or tomorrow (reinstalling MacOS right now for unrelated reasons, so I can't do it this second).

I uploaded a test build [here](https://github.com/WrathofthePast/razer-macos/releases/tag/v0.4.8.test2). Let me know if everything works with a bluetooth connection. If it does, I'll open a PR to make it official.

Let me know whether it still pops up as the Basilisk Ultimate still when using the receiver. I'm wondering whether it'll appear correctly now that it has the assets. However, my guess is it will not.

**Comment by tonydang4610** [comment]
Just tried it with a Bluetooth connection. It does not show up, unfortunately.

**Comment by WrathofthePast** [comment]
> Just tried it with a Bluetooth connection. It does not show up, unfortunately.

Did you try pressing "Refresh Device List"? If so, what macOS version are you on and are you on an M1 or Intel Mac?

**Comment by tonydang4610** [comment]
1. Yes
2. 12.2 Beta
3. M1

**Comment by WrathofthePast** [comment]
> 1. Yes

Do you have other Razer devices? If so, do they appear with the build I provided?

> 2. 12.2 Beta

Beta 1 or 2? Apple released Developer Beta 2 on Tuesday & Public Beta 2 on Wednesday.

I tested it on Beta 2, so this is a necessary distinction.

> 3. M1

I tested it on Intel & we've had issues with M1 in the past. Are you able to test it on an Intel Mac?

> Let me know whether it still pops 
> up as the Basilisk Ultimate still 
> when using the receiver. I'm 
> wondering whether it'll appear 
> correctly now that it has the 
> assets. However, my guess is it 
> will not.

What about this?

**Comment by CodeMan98038** [comment]
Hey, I downloaded and got the connection. it shows up in device list and lets me change light settings, however the mouse doesn't change. Do you know how I can make it so it does? Thanks

**Comment by tonydang4610** [comment]
> > 1. Yes
> 
> Do you have other Razer devices? If so, do they appear with the build I provided?
> 
> > 2. 12.2 Beta
> 
> Beta 1 or 2? Apple released Developer Beta 2 on Tuesday & Public Beta 2 on Wednesday.
> 
> I tested it on Beta 2, so this is a necessary distinction.
> 
> > 3. M1
> 
> I tested it on Intel & OpenRazer claims they still don't have it working. Are you able to test it on an Intel Mac to remove variables?
> 
> > Let me know whether it still pops
> > up as the Basilisk Ultimate still
> > when using the receiver. I'm
> > wondering whether it'll appear
> > correctly now that it has the
> > assets. However, my guess is it
> > will not.
> 
> What about this?

UPDATE: The mouse does register and works when using the dongle. But for Bluetooth, it still does not register. I'm on 12.2 Dev Beta 2. It does not pop up as the Basilisk Ultimate. It pops up as its correct model name, Razer Basilisk X Hyperspeed.

**Comment by CodeMan98038** [comment]
> > > 1. Yes
> > 
> > 
> > Do you have other Razer devices? If so, do they appear with the build I provided?
> > > 2. 12.2 Beta
> > 
> > 
> > Beta 1 or 2? Apple released Developer Beta 2 on Tuesday & Public Beta 2 on Wednesday.
> > I tested it on Beta 2, so this is a necessary distinction.
> > > 3. M1
> > 
> > 
> > I tested it on Intel & OpenRazer claims they still don't have it working. Are you able to test it on an Intel Mac to remove variables?
> > > Let me know whether it still pops
> > > up as the Basilisk Ultimate still
> > > when using the receiver. I'm
> > > wondering whether it'll appear
> > > correctly now that it has the
> > > assets. However, my guess is it
> > > will not.
> > 
> > 
> > What about this?
> 
> UPDATE: The mouse does register and works when using the dongle. But for Bluetooth, it still does not register. I'm on 12.2 Dev Beta 2. It does not pop up as the Basilisk Ultimate. It pops up as its correct model name, Razer Basilisk X Hyperspeed.



Hey, what do you mean by the dangle, like is there a dangle that was supposed to come with my mouse? And is there any other way to let it connect and allow color changes?

**Comment by CodeMan98038** [comment]
> > > 1. Yes
> > 
> > 
> > Do you have other Razer devices? If so, do they appear with the build I provided?
> > > 2. 12.2 Beta
> > 
> > 
> > Beta 1 or 2? Apple released Developer Beta 2 on Tuesday & Public Beta 2 on Wednesday.
> > I tested it on Beta 2, so this is a necessary distinction.
> > > 3. M1
> > 
> > 
> > I tested it on Intel & OpenRazer claims they still don't have it working. Are you able to test it on an Intel Mac to remove variables?
> > > Let me know whether it still pops
> > > up as the Basilisk Ultimate still
> > > when using the receiver. I'm
> > > wondering whether it'll appear
> > > correctly now that it has the
> > > assets. However, my guess is it
> > > will not.
> > 
> > 
> > What about this?
> 
> UPDATE: The mouse does register and works when using the dongle. But for Bluetooth, it still does not register. I'm on 12.2 Dev Beta 2. It does not pop up as the Basilisk Ultimate. It pops up as its correct model name, Razer Basilisk X Hyperspeed.

I did connect mine with the dongle, and now it is connected. It lets me change the DPI, polling rate, and color. However when I use the color changer, it doesn't do anything to my mouse. Is there something I'm doing wrong?

**Comment by tonydang4610** [comment]
> > > > 1. Yes
> > > 
> > > 
> > > Do you have other Razer devices? If so, do they appear with the build I provided?
> > > > 2. 12.2 Beta
> > > 
> > > 
> > > Beta 1 or 2? Apple released Developer Beta 2 on Tuesday & Public Beta 2 on Wednesday.
> > > I tested it on Beta 2, so this is a necessary distinction.
> > > > 3. M1
> > > 
> > > 
> > > I tested it on Intel & OpenRazer claims they still don't have it working. Are you able to test it on an Intel Mac to remove variables?
> > > > Let me know whether it still pops
> > > > up as the Basilisk Ultimate still
> > > > when using the receiver. I'm
> > > > wondering whether it'll appear
> > > > correctly now that it has the
> > > > assets. However, my guess is it
> > > > will not.
> > > 
> > > 
> > > What about this?
> > 
> > 
> > UPDATE: The mouse does register and works when using the dongle. But for Bluetooth, it still does not register. I'm on 12.2 Dev Beta 2. It does not pop up as the Basilisk Ultimate. It pops up as its correct model name, Razer Basilisk X Hyperspeed.
> 
> I did connect mine with the dongle, and now it is connected. It lets me change the DPI, polling rate, and color. However when I use the color changer, it doesn't do anything to my mouse. Is there something I'm doing wrong?

The mouse does not have any RGB. The color changer is useless for the mouse.

**Comment by CodeMan98038** [comment]
oh thank you! I was confused lol

**Comment by WrathofthePast** [comment]
> I did connect mine with the dongle, and now it is connected. It lets me change the DPI, polling rate, and color. However 
> when I use the color changer, it doesn't do anything to my mouse. Is there something I'm doing wrong?

@CodeMan98038, what specific color options do you see? You shouldn't be able to see any as I listed them all as missing features on the frontend. I either forgot something or it could be related to the next section.

> Just tried it with a Bluetooth connection. It does not show up, unfortunately.

@tonydang4610, I found the issue. The screenshot provided by @chifle-silva used the receiver connection which has a different device-id. Can either of you provide the same screenshot when using a bluetooth connection (it'll be listed under the bluetooth submenu instead of USB)?

This is also why it correctly appears now when using the receiver... it's overwriting the Basilisk Ultimate Receiver.

**Comment by tonydang4610** [comment]
<img width="224" alt="Screenshot at Jan 17 13-41-51" src="https://user-images.githubusercontent.com/94892948/149828917-a81803cb-a1f8-4871-a51d-dd2fdff78c02.png">

**Comment by WrathofthePast** [comment]
> <img alt="Screenshot at Jan 17 13-41-51" width="224" src="https://user-images.githubusercontent.com/94892948/149828917-a81803cb-a1f8-4871-a51d-dd2fdff78c02.png">

Yep, that was it. The product-id for the receiver was 0x0083. I'll upload a new test release in a moment.

**Comment by chifle-silva** [comment]
So, I screwed up? Sorry for being away, I was just catching up and testing. Let me know if I can help you

**Comment by chifle-silva** [comment]
Makes sense your comments... with the dongle I see the device appearing as bluetooth and if I remove the dongle I don't see anything

**Comment by WrathofthePast** [comment]
@chifle-silva, @CodeMan98038, @tonydang4610, I uploaded a new test release [here](https://github.com/WrathofthePast/razer-macos/releases/tag/v0.4.8-test3). I put all the changes in the release notes.

Let me know if everything works correctly now for **both** connections. Also, let me know if any color options still appear.

**Comment by tonydang4610** [comment]
Connected my mouse via Bluetooth and it still does not show up. I tried refreshing the device list as well.

**Comment by WrathofthePast** [comment]
> Connected my mouse via Bluetooth and it still does not show up. I tried refreshing the device list as well.

What about connecting via the receiver? Does it show "Basilisk X Hyperspeed (Receiver)" now? Also, does it list any color options?

**Comment by tonydang4610** [comment]
> > Connected my mouse via Bluetooth and it still does not show up. I tried refreshing the device list as well.
> 
> What about connecting via the receiver? Does it show "Basilisk X Hyperspeed (Receiver)" now? Also, does it list any color options?

Connecting via the receiver works perfectly. It does still list color options, yes.

**Comment by WrathofthePast** [comment]
> > > Connected my mouse via Bluetooth and it still does not show up. I tried refreshing the device list as well.
> > 
> > 
> > What about connecting via the receiver? Does it show "Basilisk X Hyperspeed (Receiver)" now? Also, does it list any color options?
> 
> Connecting via the receiver works perfectly. It does still list color options, yes.

Which color options?

**Comment by tonydang4610** [comment]
<img width="501" alt="color options" src="https://user-images.githubusercontent.com/94892948/149832576-643729ae-1cfe-4073-914c-8d54ad6f1a3b.png">

**Comment by WrathofthePast** [comment]
> <img alt="color options" width="501" src="https://user-images.githubusercontent.com/94892948/149832576-643729ae-1cfe-4073-914c-8d54ad6f1a3b.png">

What about the menu bar tray? Does that list any color options still?

Also, does changing the DPI & polling rate work?

**Comment by tonydang4610** [comment]
<img width="312" alt="menu bar" src="https://user-images.githubusercontent.com/94892948/149832947-44875ec5-c8fe-4c99-bf9b-29812aa6f5d1.png">
Changing the DPI and polling rate works fine.

**Comment by chifle-silva** [comment]
> Connected my mouse via Bluetooth and it still does not show up. I tried refreshing the device list as well.

same, via bluetooth does not show up. With the receiver it looks correct.

**Comment by WrathofthePast** [comment]
> <img alt="menu bar" width="312" src="https://user-images.githubusercontent.com/94892948/149832947-44875ec5-c8fe-4c99-bf9b-29812aa6f5d1.png">
> 
> Changing the DPI and polling rate works fine.

@chifle-silva, @tonydang4610, I found why most color options were still appearing... here's an [updated test build](https://github.com/WrathofthePast/razer-macos/releases/tag/v0.4.8-test4) with it fixed.

Let me know if that issue is fixed. Also, let me know if the custom color creation tool still appears when you click on the name. Lastly, try testing bluetooth mode again. I'm hoping by fixing this issue, bluetooth will work.

**Comment by chifle-silva** [comment]
Bluetooth still not working. 

With the dongle the colors were removed but when you click on the device I see all the menu in black:


<img width="507" alt="Screen Shot 2022-01-17 at 21 47 49" src="https://user-images.githubusercontent.com/8663001/149835493-0602a954-f31e-46fc-99a2-dbf6fb318aa1.png">

**Comment by WrathofthePast** [comment]
> Bluetooth still not working.
> 
> With the dongle the colors were removed but when you click on the device I see all the menu in black:
> 
> <img alt="Screen Shot 2022-01-17 at 21 47 49" width="507" src="https://user-images.githubusercontent.com/8663001/149835493-0602a954-f31e-46fc-99a2-dbf6fb318aa1.png">

I was afraid of that. I wasn't sure whether the "none" arg removes the none color option or disables all features. Turns out it was the latter. Here's an [updated release](https://github.com/WrathofthePast/razer-macos/releases/tag/v0.4.8-test5) once again. Let me know the same things as last time including retesting bluetooth.

**Comment by chifle-silva** [comment]
> > Bluetooth still not working.
> > With the dongle the colors were removed but when you click on the device I see all the menu in black:
> > <img alt="Screen Shot 2022-01-17 at 21 47 49" width="507" src="https://user-images.githubusercontent.com/8663001/149835493-0602a954-f31e-46fc-99a2-dbf6fb318aa1.png">
> 
> I was afraid of that. I wasn't sure whether the "none" arg removes the none color option or disables all features. Turns out it was the latter. Here's an [updated release](https://github.com/WrathofthePast/razer-macos/releases/tag/v0.4.8-test5) once again. Let me know the same things as last time including retesting bluetooth.

still the same for me... black menu, bluetooth not showing up

**Comment by WrathofthePast** [comment]
> > > Bluetooth still not working.
> > > With the dongle the colors were removed but when you click on the device I see all the menu in black:
> > > <img alt="Screen Shot 2022-01-17 at 21 47 49" width="507" src="https://user-images.githubusercontent.com/8663001/149835493-0602a954-f31e-46fc-99a2-dbf6fb318aa1.png">
> > 
> > 
> > I was afraid of that. I wasn't sure whether the "none" arg removes the none color option or disables all features. Turns out it was the latter. Here's an [updated release](https://github.com/WrathofthePast/razer-macos/releases/tag/v0.4.8-test5) once again. Let me know the same things as last time including retesting bluetooth.
> 
> still the same for me... black menu, bluetooth not showing up

Did you remember to press "refresh device list" for the receiver mode?

**Comment by chifle-silva** [comment]
yes, I did... still having issues

**Comment by WrathofthePast** [comment]
> yes, I did... still having issues

I found something interesting. In #463, they had the same issue with a different mouse... I looked what the two had in common and that's "breathe," "reactive," & "oldMouseEffects" args in featuresMissing. I know from my own devices that the "oldMouseEffects" arg works as intended, so that narrows it down to "reactive" and/or "breathe" args.

Here's a [new test release](https://github.com/WrathofthePast/razer-macos/releases/tag/v0.4.8-test6) where I deleted those two args. Note that I only did this for the receiver connection, so don't bother checking bluetooth yet. If this fixes it, we'll add one of them back to see if it stays fixed. Either way, that tells us which one is the issue.

**Comment by chifle-silva** [comment]
> > yes, I did... still having issues
> 
> I found something interesting. In #463, they have the same issue on a different mouse... I looked what the two had in common and that's "breathe," "reactive," & "oldMouseEffects" args in featuresMissing. I know from my own devices that the "oldMouseEffects" arg works as intended, so that narrows it down to "reactive" and/or "breathe" args.
> 
> Here's a [new test release](https://github.com/WrathofthePast/razer-macos/releases/tag/v0.4.8-test6) where I deleted those two args. Note that I only did this for the receiver connection, so don't bother checking bluetooth yet. If this fixes it, we'll add one of them back to see if it stays fixed. Either way, that tells us which one is the issue.

Menu still back, when selecting the device. Breath and reactive are there but I don't think they apply to it, or not sure how to...


<img width="859" alt="Screen Shot 2022-01-17 at 22 38 21" src="https://user-images.githubusercontent.com/8663001/149839700-484f386c-d17c-4f6f-b558-7cd7b8a8e4fe.png">

**Comment by WrathofthePast** [comment]
> > > yes, I did... still having issues
> > 
> > 
> > I found something interesting. In #463, they have the same issue on a different mouse... I looked what the two had in common and that's "breathe," "reactive," & "oldMouseEffects" args in featuresMissing. I know from my own devices that the "oldMouseEffects" arg works as intended, so that narrows it down to "reactive" and/or "breathe" args.
> > Here's a [new test release](https://github.com/WrathofthePast/razer-macos/releases/tag/v0.4.8-test6) where I deleted those two args. Note that I only did this for the receiver connection, so don't bother checking bluetooth yet. If this fixes it, we'll add one of them back to see if it stays fixed. Either way, that tells us which one is the issue.
> 
> Menu still back, when selecting the device. Breath and reactive are there but I don't think they apply to it, or not sure how to...

Very odd that the window is still blank. Here's yet another [release](https://github.com/WrathofthePast/razer-macos/releases/tag/v0.4.8-test6) (same link as last time; I edited the existing release and updated the release notes). This time I removed the "static" & "spectrum" args as well, so all possible culprits are removed. 

We know that it has something to do with the missingFeatures args because the DPI & polling rate populated before when I had the typo on the first six of eight. We know it's not the last two for that reason. Also, I know that it's not "waveSimple" from my own mice. That leaves five and I've previously removed three of those.

**Comment by WrathofthePast** [comment]
@tonydang4610, @chifle-silva, did you try the new [test release](https://github.com/WrathofthePast/razer-macos/releases/tag/v0.4.8-test6)? We're close to completing receiver mode. If we can get this done quickly, it can be in the next official release.

**Comment by chifle-silva** [comment]
Just tested it. With the receiver seems to be working fine. I see the device and the menu.
With the bluetooth still does not work, but since I am using it with the receiver I am more than ok with it

**Comment by tonydang4610** [comment]
<img width="313" alt="image" src="https://user-images.githubusercontent.com/94892948/150650126-112fe125-9aff-4088-a515-98d3d2cbb34a.png">
I'm still seeing options for color modes.

**Comment by tonydang4610** [comment]
> <img alt="image" width="313" src="https://user-images.githubusercontent.com/94892948/150650126-112fe125-9aff-4088-a515-98d3d2cbb34a.png">
> 
> I'm still seeing options for color modes.

The window is also no longer blank and works fine.

**Comment by dhobi** [comment]
I think bluetooth is out of scope here as the library (afair) only lists usb devices.

**Comment by WrathofthePast** [comment]
> The window is also no longer blank and works fine.

@tonydang4610, @chifle-silva, that means the culprit was `static` or `spectrum`. Here's an updated [version](https://github.com/WrathofthePast/razer-macos/releases/tag/v0.4.8-test7). I readded `breathe`, `reactive`, & `spectrum` to featuresMissing. Lmk if the window still populates.

> I think bluetooth is out of scope here as the library (afair) only lists usb devices.

@dhobi, openrazer got some bluetooth functionality working on Razer Atheris a couple years ago which is supported by razer-macos. I'm hoping that once I get this current bug worked out, then apply the fix to bluetooth, it'll work. However, I don't have high hopes of that working.

**Comment by WrathofthePast** [comment]
@chifle-silva, @tonydang4610,  I found the bug. I forgot a line in the backend. Here's a new [release](https://github.com/WrathofthePast/razer-macos/releases/tag/v0.4.8-test8). Hopefully, it'll work as intended now and maybe even bluetooth will work.

**Comment by WrathofthePast** [comment]
@chifle-silva, @tonydang4610, I spoke to openrazer and they explained the intermittent blank window issue and provided the code to fix it. Here's a new [test release](https://github.com/WrathofthePast/razer-macos/releases/tag/v0.4.8-test9). It should work flawlessly now. Hopefully, bluetooth mode will work too, but openrazer wasn't able to get that working themselves, so I doubt it'll happen.

**Comment by tonydang4610** [comment]
I still see a blank window when opening it up.

**Comment by chifle-silva** [comment]
> @chifle-silva, @tonydang4610, I spoke to openrazer and they explained the intermittent blank window issue and provided the code to fix it. Here's a new [test release](https://github.com/WrathofthePast/razer-macos/releases/tag/v0.4.8-test9). It should work flawlessly now. Hopefully, bluetooth mode will work too, but openrazer wasn't able to get that working themselves, so I doubt it'll happen.

I am getting a blank window.

**Comment by WrathofthePast** [comment]
> I'll provide more detail:
> 
> 
> 
> macOS: Monterey 12.3
> 
> App version: The linked test release [0.4.8test9](https://github.com/WrathofthePast/razer-macos/releases/tag/v0.4.8-test9)
> 
> 
> 
> Observations:
> 
> * The mouse is detected, but seemingly only while connected to the receiver
> 
> * The battery meter seems to be accurate (within 1%? Bluetooth reported 9% while the app reports 8%, good enough for me!)
> 
> * When clicking on the mouse from the menu, I get a blank window regardless of the connection status of the mouse
> 
> * I'm not sure if it's desired, but when the mouse is completely disconnected, it still showed up in the list regardless of clearing settings or refreshing devices
> 
> 
> 
> Logs from Console:
> 
> ```
> 
> default	21:17:26.546643-0400	Razer macOS	SetFrontProcess: asn=0x0-0x2b12b1 options=1
> 
> default	21:17:26.554048-0400	runningboardd	Acquiring assertion targeting [app<application.com.electron.razer-macos.16393977.16394170(501)>:9946] from originator [daemon<com.apple.coreservices.launchservicesd>:145] with description <RBSAssertionDescriptor| "frontmost:9946" ID:213-145-19621 target:9946 attributes:[
> 
> 	<RBSDomainAttribute| domain:"com.apple.launchservicesd" name:"RoleUserInteractiveFocal" sourceEnvironment:"(null)">
> 
> 	]>
> 
> default	21:17:26.554328-0400	runningboardd	Assertion 213-145-19621 (target:[app<application.com.electron.razer-macos.16393977.16394170(501)>:9946]) will be created as active
> 
> default	21:17:26.555779-0400	NowPlayingTouchUI	[NowPlayingTouchUI] Reloading available apps due to workspace activating an application: NSConcreteNotification 0x600001dac3c0 {name = NSWorkspaceDidActivateApplicationNotification; object = <NSWorkspace: 0x600001f9c0f0>; userInfo = {
> 
>     NSWorkspaceApplicationKey = "<NSRunningApplication: 0x6000033c2380 (com.electron.razer-macos - 9946) LSASN:{hi=0x0;lo=0x2b12b1}>";
> 
> }}
> 
> default	21:17:26.557322-0400	runningboardd	Calculated state for app<application.com.electron.razer-macos.16393977.16394170(501)>: running-active (role: UserInteractiveFocal)
> 
> default	21:17:26.557883-0400	runningboardd	[app<application.com.electron.razer-macos.16393977.16394170(501)>:9946] Ignoring jetsam update because this process is not memory-managed
> 
> default	21:17:26.559161-0400	runningboardd	[app<application.com.electron.razer-macos.16393977.16394170(501)>:9946] Ignoring suspend because this process is not lifecycle managed
> 
> default	21:17:26.560208-0400	runningboardd	[app<application.com.electron.razer-macos.16393977.16394170(501)>:9946] Set darwin role to: UserInteractiveFocal
> 
> default	21:17:26.560987-0400	runningboardd	[app<application.com.electron.razer-macos.16393977.16394170(501)>:9946] Ignoring GPU update because this process is not GPU managed
> 
> default	21:17:26.562160-0400	runningboardd	Acquiring assertion targeting [app<application.com.electron.razer-macos.16393977.16394170(501)>:9946] from originator [daemon<com.apple.coreservices.launchservicesd>:145] with description <RBSAssertionDescriptor| "notification:9946" ID:213-145-19622 target:9946 attributes:[
> 
> 	<RBSDomainAttribute| domain:"com.apple.launchservicesd" name:"LSNotification" sourceEnvironment:"(null)">
> 
> 	]>
> 
> default	21:17:26.562788-0400	runningboardd	Assertion 213-145-19622 (target:[app<application.com.electron.razer-macos.16393977.16394170(501)>:9946]) will be created as active
> 
> default	21:17:26.565464-0400	runningboardd	[app<application.com.electron.razer-macos.16393977.16394170(501)>:9946] Ignoring jetsam update because this process is not memory-managed
> 
> default	21:17:26.565664-0400	runningboardd	[app<application.com.electron.razer-macos.16393977.16394170(501)>:9946] Ignoring suspend because this process is not lifecycle managed
> 
> default	21:17:26.565864-0400	runningboardd	[app<application.com.electron.razer-macos.16393977.16394170(501)>:9946] Ignoring GPU update because this process is not GPU managed
> 
> default	21:17:26.565381-0400	runningboardd	Calculated state for app<application.com.electron.razer-macos.16393977.16394170(501)>: running-active (role: UserInteractiveFocal)
> 
> default	21:17:26.600893-0400	Razer macOS	[0x7fe50b869600] CVDisplayLinkStart
> 
> default	21:17:26.600948-0400	Razer macOS	[0x7fe50b869620] CVDisplayLink::start
> 
> default	21:17:26.601083-0400	Razer macOS	[0x600003734770] CVXTime::reset
> 
> default	21:17:26.620306-0400	Razer macOS	[0x7fe50b869600] CVDisplayLinkStop
> 
> default	21:17:26.620363-0400	Razer macOS	[0x7fe50b869620] CVDisplayLink::stop
> 
> default	21:17:31.757301-0400	runningboardd	Assertion did invalidate due to timeout: 213-145-19622 (target:[app<application.com.electron.razer-macos.16393977.16394170(501)>:9946])
> 
> default	21:17:31.873636-0400	runningboardd	[app<application.com.electron.razer-macos.16393977.16394170(501)>:9946] Ignoring jetsam update because this process is not memory-managed
> 
> default	21:17:31.873716-0400	runningboardd	[app<application.com.electron.razer-macos.16393977.16394170(501)>:9946] Ignoring suspend because this process is not lifecycle managed
> 
> default	21:17:31.873755-0400	runningboardd	[app<application.com.electron.razer-macos.16393977.16394170(501)>:9946] Ignoring GPU update because this process is not GPU managed
> 
> default	21:17:31.874057-0400	runningboardd	Calculated state for app<application.com.electron.razer-macos.16393977.16394170(501)>: running-active (role: UserInteractiveFocal)
> 
> default	21:17:34.792747-0400	Razer macOS	SetFrontProcess: asn=0x0-0x2b12b1 options=1
> 
> default	21:17:47.373955-0400	Razer macOS	SetFrontProcess: asn=0x0-0x2b12b1 options=1
> 
> default	21:17:47.392826-0400	Razer macOS	[0x7fe50b869600] CVDisplayLinkStart
> 
> default	21:17:47.392869-0400	Razer macOS	[0x7fe50b869620] CVDisplayLink::start
> 
> default	21:17:47.392981-0400	Razer macOS	[0x600003734770] CVXTime::reset
> 
> default	21:17:47.403909-0400	Razer macOS	[0x7fe50b869600] CVDisplayLinkStop
> 
> default	21:17:47.403954-0400	Razer macOS	[0x7fe50b869620] CVDisplayLink::stop
> 
> default	21:17:51.844389-0400	runningboardd	Invalidating assertion 213-145-19621 (target:[app<application.com.electron.razer-macos.16393977.16394170(501)>:9946]) from originator [daemon<com.apple.coreservices.launchservicesd>:145]
> 
> default	21:17:51.963506-0400	runningboardd	[app<application.com.electron.razer-macos.16393977.16394170(501)>:9946] Ignoring jetsam update because this process is not memory-managed
> 
> default	21:17:51.963512-0400	runningboardd	Calculated state for app<application.com.electron.razer-macos.16393977.16394170(501)>: running-active (role: UserInteractive)
> 
> default	21:17:51.963548-0400	runningboardd	[app<application.com.electron.razer-macos.16393977.16394170(501)>:9946] Ignoring suspend because this process is not lifecycle managed
> 
> default	21:17:51.963682-0400	runningboardd	[app<application.com.electron.razer-macos.16393977.16394170(501)>:9946] Set darwin role to: UserInteractive
> 
> default	21:17:51.963731-0400	runningboardd	[app<application.com.electron.razer-macos.16393977.16394170(501)>:9946] Ignoring GPU update because this process is not GPU managed
> 
> ```

Thank you for this information. I'll make a new test release soon. I should be able to fix it. However, Bluetooth is unlikely to work, so I won't include it. We can look into that after receiver support is complete.

**Comment by err53** [comment]
Hi everyone, any updates on this feature?
